chain/get_ancestors.go

```diff
@@ -51,37 +51,32 @@ func GetRecentAncestors(ctx context.Context, base types.TipSet, provider TipSetP
 		earliestAncestorHeight = types.NewBlockHeight(uint64(0))
 	}
 
-	// Step 1 -- gather all tipsets with a height greater than the earliest
-	// possible proving period start still in scope for the given head.
 	iterator := IterAncestors(ctx, provider, base)
+	// Step 1 -- gather all tipsets up to the first tipset with a height less than
+	// or equal to the earliest possible proving period start
 	provingPeriodAncestors, err := CollectTipSetsOfHeightAtLeast(ctx, iterator, earliestAncestorHeight)
 	if err != nil {
 		return nil, err
 	}
-	firstExtraRandomnessAncestorsCids, err := provingPeriodAncestors[len(provingPeriodAncestors)-1].Parents()
-	if err != nil {
-		return nil, err
-	}
-	// no parents means hit genesis so return the whole chain
-	if firstExtraRandomnessAncestorsCids.Len() == 0 {
-		return provingPeriodAncestors, nil
-	}
 
-	// Step 2 -- gather the lookback tipsets directly preceding provingPeriodAncestors.
-	lookBackTS, err := provider.GetTipSet(firstExtraRandomnessAncestorsCids)
-	if err != nil {
-		return nil, err
+	// Step 2 -- skip the next tipset which added if iterator not complete
+	if !iterator.Complete() {
+		err = iterator.Next()
+		if err != nil {
+			return nil, err
+		}
 	}
-	iterator = IterAncestors(ctx, provider, lookBackTS)
+
+	// Step 3 -- gather the lookback tipsets directly preceding provingPeriodAncestors.
 	extraRandomnessAncestors, err := CollectAtMostNTipSets(ctx, iterator, lookback)
 	if err != nil {
 		return nil, err
 	}
 	return append(provingPeriodAncestors, extraRandomnessAncestors...), nil
 }
 
-// CollectTipSetsOfHeightAtLeast collects all tipsets with a height greater
-// than or equal to minHeight from the input tipset.
+// CollectTipSetsOfHeightAtLeast collects all tipsets up to the first tipset with a height less than
+// or equal to the earliest possible proving period start
```

Comment:
"up to and including"

Two blocking issues:
- The name of this function is now misleading. Maybe `CollectTipSetsPastHeight`? Not sure - it's hard to name this behaviour
- This function has another caller in `chain/traversal.go` and this behaviour changes appears to be incompatible with its needs (excluding the equal-height block). Unfortunately, `CollectTipsToCommonAncestor` does not have direct unit tests.

Some suggestions for resolution:
- Leave this function alone and write more code in `GetRecentAncestors` to traverse the additional block
- Leave this function alone and write a new one for the needs of `GetRecentAncestors`
- Revert this change, then replace `minheight` with a predicate and rename to `CollectTipSetsWhile`.